tests/test_model_tool.py

```diff
@@ -0,0 +1,158 @@
+
+
+from json import loads as json_loads
+from os.path import join
+from unittest import TestCase
+
+from sortedcontainers import SortedSet
+
+from cyclonedx.exception.model import MutuallyExclusivePropertiesException
+from cyclonedx.model.bom import Bom
+from cyclonedx.model.component import Component
+from cyclonedx.model.contact import OrganizationalEntity
+from cyclonedx.model.service import Service
+from cyclonedx.model.tool import Tool, ToolsRepository, ToolsRepositoryHelper
+from cyclonedx.output.json import JsonV1Dot5
+from cyclonedx.output.xml import XmlV1Dot5
+from tests import OWN_DATA_DIRECTORY, reorder
+from tests._data.models import get_bom_with_tools_with_component_and_service
+
+
+class TestModelTool(TestCase):
+
+    def test_sort(self) -> None:
+        # expected sort order: (vendor, name, version)
+        expected_order = [0, 1, 2, 3, 4, 5, 6]
+        tools = [
+            Tool(vendor='a', name='a', version='1.0.0'),
+            Tool(vendor='a', name='a', version='2.0.0'),
+            Tool(vendor='a', name='b', version='1.0.0'),
+            Tool(vendor='a', name='b'),
+            Tool(vendor='b', name='a'),
+            Tool(vendor='b', name='b', version='1.0.0'),
+            Tool(name='b'),
+        ]
+        sorted_tools = sorted(tools)
+        expected_tools = reorder(tools, expected_order)
+        self.assertListEqual(sorted_tools, expected_tools)
+
+    def test_tool_with_component_and_service_load_json(self) -> None:
+        test_file = join(OWN_DATA_DIRECTORY, 'json', '1.5',
+                         'bom_with_tool_with_component_and_service.json')
+        with open(test_file, encoding='UTF-8') as f:
+            bom_json = json_loads(f.read())
+        bom = Bom.from_json(bom_json)   # type: ignore[attr-defined]
+        good_bom = get_bom_with_tools_with_component_and_service()
+        self.assertTrue(bom == good_bom)
+
+    def test_tool_with_component_and_service_render_json(self) -> None:
+        bom = get_bom_with_tools_with_component_and_service()
+        test_file = join(OWN_DATA_DIRECTORY, 'json', '1.5',
+                         'bom_with_tool_with_component_and_service.json')
+        with open(test_file, encoding='utf-8') as f:
+            self.assertEqual(JsonV1Dot5(bom).output_as_string(indent=2), f.read())
+
+    def test_tool_with_component_and_service_load_xml(self) -> None:
+        test_file = join(OWN_DATA_DIRECTORY, 'xml', '1.5',
+                         'bom_with_tool_with_component_and_service.xml')
+        with open(test_file, encoding='utf-8') as s:
+            bom = Bom.from_xml(s)  # type: ignore[attr-defined]
+        self.assertEqual(bom.metadata.tools.components[0].type, 'application')
+        self.assertEqual(bom.metadata.tools.components[0].name, 'test-component')
+        self.assertEqual(bom.metadata.tools.services[0].name, 'test-service')
+
+    def test_tool_with_componet_and_service_render_xml(self) -> None:
```

Comment:
this is not the place where we do xml/json serualization.
these integration tests go here: `tests/test_output_{json,xml}.py` 

----

and you may find, that your data models are require some "migration' - so please add the suffix `_migrate` to them.
this will prevent them from being deserialized and checked - which probably will not match the original data models, whwn serialized from CDX  < 1.5

Reply:
> these integration tests go here: tests/test_output_{json,xml}.py

Can you clarify? I don't see any other render/load tests in those files. Are you saying I should move those tests to those files? Or that I should remove my tests? Or, are you saying I can delete all four of my render/load tests and the same tests will be run because I have a `get_bom_with_tools_with_component_and_service` function (which I should rename to `get_bom_with_tools_with_component_and_service_migrate`)?

Reply:
> this will prevent them from being deserialized and checked

It seems I would need to keep the deserialization checks. Is that right?